### PR TITLE
refact:  to function component

### PR DESCRIPTION
**Pull request by ChaiWithJai**
This PR is part of a larger block of work #8769 to migrate away from `connect` to use React hooks.

This PR is refactoring the `A11yIntention` component from a class-based component to function component.

### CODE REVIEW

File: src/devtools/client/debugger/src/components/A11yIntention.js [hunk 1]
```diff
@@ -3,28 +3,22 @@
...
 
-export default class A11yIntention extends React.Component {
-  state = { keyboard: false };
+export default function A11yIntention({ children }) {
+  const [isKeyboard, setIsKeyboard] = useState(false);
 
-  handleKeyDown = () => {
-    this.setState({ keyboard: true });
-  };
+  handleKeyDown = () => setIsKeyboard(true);
```
Comment (markerikson) [inline src/devtools/client/debugger/src/components/A11yIntention.js:11]: :hand: Both of these callbacks probably need a `const ` in front.

File: src/devtools/client/debugger/src/components/A11yIntention.js [hunk 2]
```diff
@@ -3,28 +3,22 @@
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
 
 //
-import React from "react";
+import React, { useState } from "react";
```
Comment (markerikson) [inline src/devtools/client/debugger/src/components/A11yIntention.js:6]: :hand: While we're at it, bonus request: can you convert this file to TypeScript?

Looking at it, I'm pretty sure that _all_ you need to do is:

- Rename it to `.tsx`
- Add a type declaration for the props that includes `children` (probably `{children: React.ReactNode}`), and use that

**Review by markerikson — CHANGES_REQUESTED** [review]
Good start! Can you fix the callback declarations, and also try converting it to TypeScript while you're at it?

File: src/devtools/client/debugger/src/components/A11yIntention.js (hunk 2 above)
Comment (ChaiWithJai) [reply src/devtools/client/debugger/src/components/A11yIntention.js:6]: Roger that!

File: src/devtools/client/debugger/src/components/A11yIntention.js (hunk 1 above)
Comment (ChaiWithJai) [reply src/devtools/client/debugger/src/components/A11yIntention.js:11]: Good catch!

**Review by markerikson — APPROVED** [review]

**Comment by markerikson** [comment]
Hmm. Seems like the UI snapshot tool had an upload failure, shouldn't be related.

**Comment by markerikson** [comment]
Thanks!

**Pull request merged**